napari/layers/surface/normals.py

```diff
@@ -34,7 +34,8 @@ class Normals(EventedModel):
 
     mode: NormalMode = Field(NormalMode.FACE, allow_mutation=False)
     visible: bool = False
-    color: Union[str, Array[float, (3,)], Array[float, (4,)]] = 'black'
+    # TODO: check if this change in stored value is desired.
```

Comment:
What does this todo refer to? The fact that it stores the `ColorValue` rather than the string? If so, I think that's fine. The downside is maybe that you can't do `ColorValue == 'red'`...

Reply:
Yeah, I left it there to highlight the change in behavior as most other changes here don't change anything.

But your comment gave me an idea. Why can't we have `__eq__` and get the behavior you described. So here's my attempt at that.

```python
import numpy as np
from pydantic import BaseModel
from napari.utils.colormaps.standardize_color import transform_color

class ColorValue(np.ndarray):
    def __new__(cls, value): 
        array = transform_color(value)[0]
        return array.view(cls)

    @classmethod
    def __get_validators__(cls):
        yield cls.__new__

    @classmethod
    def _validate(cls, value):
        return cls(value)

    def __eq__(self, other):
        return np.array_equal(self, ColorValue(other))

class MyModel(BaseModel):
    color: ColorValue

c = ColorValue('red')
c # => ColorValue([1., 0., 0., 1.], dtype=float32)
c == 'red' # => True
c == [1, 0, 0] # => True
c == 'blue' # => False

m = MyModel(c='red')
m.c # => ColorValue([1., 0., 0., 1.], dtype=float32)
m.c == 'red' # => True

```

Kinda nice, right? What do you think?

I'm pretty new to messing around with `__new__` and came up with this after reading [the numpy docs on subclassing `ndarray`](https://numpy.org/doc/stable/user/basics.subclassing.html), so there could definitely be some problem I don't see here or maybe some weird potential interactions with pydantic. I did find that yielding `cls.__new__` directly from `__get_validators__` caused a pydantic validation error with the message `Cannot change data-type for object array. (type=type_error)`, which is why I used `_validate` instead.

Reply:
Defining `__eq__` could also play nicely with `EventedModel` and I think would negate the need to add custom equality operators as we do now.

Reply:
This is awesome!

Reply:
Also curious to get @tlambert03 's opinion on the above before potentially including this here and making it a non-draft PR.

Reply:
I love this too! ❤️